use-case-zeit-now-ci-cd.md

```diff
@@ -0,0 +1,246 @@
+
+Continuous integration pipeline
+-------------------------------
+
+``` yaml
+# .semaphore/semaphore.yml
+
+# Use the latest stable version of Semaphore 2.0 YML syntax:
+version: v1.0
+
+# Name your pipeline. In case you connect multiple pipelines with promotions,
+# the name will help you differentiate between, for example, a CI build phase
+# and delivery phases.
+name: Build and test Express.js app
+
+# An agent defines the environment in which your code runs.
+# It is a combination of one of available machine types and operating
+# system images.
+# See https://docs.semaphoreci.com/article/20-machine-types
+# and https://docs.semaphoreci.com/article/32-ubuntu-1804-image
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu1804
+
+# Blocks are the heart of a pipeline and are executed sequentially.
+# Each block has a task that defines one or more jobs. Jobs define the
+# commands to execute.
+# See https://docs.semaphoreci.com/article/62-concepts
+blocks:
+  - name: Install dependencies
+    task:
+      jobs:
+        - name: npm install and cache
+          commands:
+            # Get the latest version of our source code from GitHub:
+            - checkout
+
+            # Use the version of Node.js specified in .nvmrc.
+            # Semaphore provides nvm preinstalled.
+            - nvm use
+            - node --version
+            - npm --version
+
+            # Restore dependencies from cache. This command will not fail in
+            # case of a cache miss. In case of a cache hit, npm install will
+            # run very fast.
+            # For more info on caching, see https://docs.semaphoreci.com/article/68-caching-dependencies
+            - cache restore client-node-modules-$SEMAPHORE_GIT_BRANCH-$(checksum package-lock.json),client-node-modules-$SEMAPHORE_GIT_BRANCH,client-node-modules-master
+            - npm install
+
+            # Store the latest version of node modules in cache to reuse in
+            # further blocks:
+            - cache store client-node-modules-$SEMAPHORE_GIT_BRANCH-$(checksum package-lock.json) node_modules
+
+  - name: Run tests
+    task:
+      jobs:
+        - name: npm test
+          commands:
+            - checkout
+            - nvm use
+            - cache restore client-node-modules-$SEMAPHORE_GIT_BRANCH-$(checksum package-lock.json),client-node-modules-$SEMAPHORE_GIT_BRANCH,client-node-modules-master
+            - npm test
+
+promotions:
+  # Deployment to staging can be triggered manually:
+  - name: Deploy to staging
+    pipeline_file: deploy-staging.yml
+
+  # Automatically deploy to production on successful builds on master branch:
+  - name: Deploy to production
+    pipeline_file: deploy-production.yml
+    auto_promote_on:
+      - result: passed
+        branch:
+          - master
+```
+
+Just 2 blocks make the CI pipeline:
+
+![CI Pipeline](https://github.com/semaphoreci-demos/semaphore-demo-zeit-now/raw/master/images/semaphore-zeit-now-ci.png)
```

Comment:
Let's have one image of the entire workflow, with expanded deployment to production. That image should, like on other pages, appear at the end of the _Overview of the CI/CD pipeline_ section.